Feature/KAS-3082 Pub v3 Details: Decisions

mdv-professional opened the pull request:
[Jira](https://kanselarij.atlassian.net/browse/KAS-3082)

**Notes**
Aantal pagina's is nog niet geimplementeerd.

Commit:
(move) publication/documents > publication/decisions
KAS-3082 pub: v3: details: decisions

Commit:
(rename) route publication/documents > publication/decisions
KAS-3082 pub: v3: details: decisions

Commit:
add empty decisions info panel
KAS-3082 pub: v3: details: decisions

Commit:
add decision date to decisions info panel
KAS-3082 pub: v3: details: decisions

Commit:
add regulation type
KAS-3082 pub: v3: details: decisions info

Commit:
add placeholder for page count
KAS-3082 pub: v3: details: decisions info

Commit:
Merge branch 'development' into feature/KAS-3082-pub-v3-details--decisions

Commit:
implement isViaCouncilOfMinisters check
KAS-3082 pub: v3: details: decisions info

Commit:
reset on cancel
KAS-3082 pub: v3: details: decisions info

Commit:
allow clear regulation type
KAS-3082 pub: v3: details: decisions info

Commit:
(rename) update data-test attributes
KAS-3082 pub: v3: details: decisions info

Commit:
(rename) update data-test attributes
KAS-3082 pub: v3: details: decisions info

Commit:
(rename) update data-test attributes
KAS-3082 pub: v3: details: decisions

Commit:
Merge branch 'feature/KAS-3082-pub-v3-details--decisions' of github.com:kanselarij-vlaanderen/frontend-kaleidos into feature/KAS-3082-pub-v3-details--decisions

Commit:
(code-style) still prefer const
KAS-3082 pub: v3: details: decisions info

Commit:
Merge branch 'development' into feature/KAS-3082-pub-v3-details--decisions

Commit:
remove duplicate sidebar fields
KAS-3082 pub: v3: details: decisions info

Commit:
(lint-fix)
KAS-3082 pub: v3: details: decisions info

Commit:
(fix) missing page title key
KAS-3082 pub: v3: details: decisions info

Commit:
(style) no padding for loadable content
reason: It can make the component change height while loading. The reason for creating this util was exactly the opposite.
KAS-3081 pub: v3: details: info + remark

Commit:
(refactor) replace PublicationService.getIsViaCouncilOfMinisters(...) by PublicationFlow.isViaCouncilOfMinisters
KAS-3082 pub: v3: details: decisions info

ghost commented:
@brenner-company Nog een styling issue: de "clear"-knop van de PowerSelect is nog niet ideaal gestyled, eker niet als met de PowerSelect iets geselecteerd is dat op meerdere regels wordt weergegeven.

Commit:
Merge branch 'development' into feature/KAS-3082-pub-v3-details--decisions

ghost commented:
Er is mij hier nog een issue opgevallen: de styling van Utils::LoadableContent component zal ik nog eens moeten bekijken.

brenner-company commented:
@mdv-professional Kan je van beide issues (of is die 2de niet direct van toepassing op mij?) even een screenshot voorzien? Zo kan ik vlotter en correcter alles debuggen.

Commit:
Merge branch 'development' into feature/KAS-3082-pub-v3-details--decisions

ghost commented:
- [x] Ik zal dezelfde consistentie-aanpassingen maken als in de Publicatiedossier-info-branch.
- [ ] Styling van de PowerSelect @brenner-company:
![Screenshot from 2022-01-31 11-44-13](https://user-images.githubusercontent.com/80692792/151780209-5c210aff-15b0-4c33-97de-0aed78c5447a.png)
- [x] `Auk::LoadableContent`-issue: Dit los ik op.

brenner-company commented:
@mdv-professional Geen probleem, ik probeer dat te bekijken van zodra ik kan 👍

Commit:
consistency: naming
KAS-3082 pub: v3: details: decision info

Commit:
consistency: use model properties in template: regulation type
KAS-3082 pub: v3: details: decision info

Commit:
consistency: use model properties in template: decision date
KAS-3082 pub: v3: details: decision info

Commit:
(cleanup + eslint)
KAS-3082 pub: v3: details: decision info

Commit:
(refactor) move getIsViaCouncilOfMinisters method back to service
KAS-3082 pub: v3: details: decision info

Commit:
clearer code
KAS-3082 pub: v3: details: decision info

Commit:
clearer code
KAS-3082 pub: v3: details: decision info

Commit:
Utils::LoadableContent: adapt to height of container
KAS-3082 pub: v3: details: decision info

ghost commented:
@brenner-company Ik heb een kleine wijziging doorgevoerd in de Utils::LoadableContent. De reden was om te verhelpen dat de grootte van het "Besluit(en) info"-paneel versprong tijdens het saven. Mogelijks weer gij hier css-gewijs een betere oplossing. Ik vond het vreemd dat op de loader-wrapper een padding zit, maar de loader horizontaal en verticaal gecentreerd wordt (De de padding lijkt dan overbodig en deze visuele bug treedt op wanneer de container kleiner is dan de loader-wrapper.

brenner-company commented:
@mdv-professional Momenteel ben ik bezig met het overzetten van die custom styling (waaronder `loadable-content`) naar `au-kaleidos-css` package en ik heb je extra aanpassing mee genomen.

brenner-company commented:
@mdv-professional Styling van de PowerSelect heb ik ondertussen ook al in orde gezet. Die aanpassing gaat mee met de PR voor https://kanselarij.atlassian.net/browse/KAS-3118.

<img width="265" alt="CleanShot 2022-02-01 at 13 38 37@2x" src="https://user-images.githubusercontent.com/18174827/151969749-59f175cd-a52e-46c4-926d-e6b8a9db222b.png">

PS: Dit is wel geen super elegante oplossing, maar meer effort lijkt me ook niet nodig aangezien ik die styling nog eens apart ga overlopen/refactoren.

erikap committed:
Don't obfuscate that publication-flow is coming from component arguments

erikap committed:
Inverse negative if-clauses for readability

erikap committed:
Inline regulation type change action in template

erikap committed:
Hide reference documents panel for publications not via minister
council

Since the decision tabs is now always visible, the documents panel
needs to be hidden for publications not via minister council

erikap committed:
Disable edit on decisions info panel if via council of ministers

erikap committed:
Add icon to navigation link on decision-info-panel

erikap committed:
Remove amount of pages

Business requirements needs to be clarified before we can implement
this feature.

erikap committed:
Update fields to be inline with design of usertest branch

erikap committed:
Merge branch 'development' into feature/KAS-3082-pub-v3-details--decisions

erikap committed:
Cleanup publication-sidebar

Leftovers from all stories where fields have moved from sidebar to info-panels

erikap committed:
Add number of pages to decision-info-panel

As discussed with the team the number of pages is currently considered
an internal working field for OVRB to estimate their workload. In the
long run the number of pages should  be automatically calculated as
the sum of all documents that are subject of the publication.

erikap committed:
Move publication-date to publication-info panel

Should have been done on a previous story. Piggybacking on this one...

erikap committed:
Only show button to open edit mode if not opened yet on info panels

erikap committed:
Fix council of minister flag in publication header

erikap committed:
Merge branch 'development' into feature/KAS-3082-pub-v3-details--decisions

erikap reviewed (approved):
Na overleg met @MikiDi en design-team is aantal pagina's voorlopig toegevoegd als nieuwe property op een publication-flow. Op lange termijn is het de bedoeling dat dit automatisch berekend wordt als de som van de documenten die voorwerp zijn van de publicatie.

Ik heb daarnaast nog wat zaken aangepast die eigenlijk al in eerdere stories/PRs hadden moeten gebeuren. Het was dus niet de verwachting dat jij die in deze PR gedaan zou hebben @mdv-professional : 
- edit icon niet tonen op info-panel als we al in edit mode zitten
- publicatiedatum vanuit sidebar verhuizen naar publication-info panel

erikap committed:
Remove kaleidos-cypress dependency from package.json

No idea how that landed there...

erikap committed:
Restore package-lock.json from development branch
